### PR TITLE
Add Section on Build and Package Management

### LINKED ISSUE #35: Document package manager

### PR DESCRIPTION
**Pull request by magnus-madsen**
Fixes #35

### CODE REVIEW

**Comment by magnus-madsen** [comment]
@mlutze If you could take a quick look and see if you have any comments :)

**Comment by magnus-madsen** [comment]
@mlutze If you could take a quick look and see if you have any comments :)

File: src/content/BuildAndPackages.js [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+                    resolution</i>, but the system is sufficient to build and share packages. There is no central
+                    package registry, so distribution and versioning must be handled manually for the moment. We propose
+                    that the semantic version of a package is included as part of its name,
+                    e.g. <Code>foo-1.2.1.fpkg</Code>.
+                </p>
+
+                <p>
+                    The Flix build system makes it easy to create a new Flix project, to compile a project, to run
+                    a project, to run tests in a project, to bundle a project into a Flix package, and to compile
+                    a project to a JAR-file.
```
Comment (mlutze) [inline src/content/BuildAndPackages.js]: Lots of `X a project` here, seems a little redundant

File: src/content/BuildAndPackages.js [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+                            <td>builds (i.e. compiles) the current project.</td>
+                        </tr>
+                        <tr>
+                            <td><Code>build-jar</Code></td>
+                            <td>builds a jar-file from the current project.</td>
+                        </tr>
+                        <tr>
+                            <td><Code>build-pkg</Code></td>
+                            <td>builds a fpkg-file from the current project.</td>
+                        </tr>
```
Comment (mlutze) [inline src/content/BuildAndPackages.js:68]: From this table alone, it's not clear what the jar and fpkg files actually are. Maybe `runnable jar file` and `consumable fpkg file`? Or something more verbose.

File: src/content/BuildAndPackages.js [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+
+                    <p>
+                        For example:
+                    </p>
+
+                    <CodeBlock>{`$ java -jar path/to/flix.jar init`}</CodeBlock>
+
+                    <p>
+                        Build commands can also be invoked from within Visual Studio Code by pressing <kbd>CTRL +
+                        P</kbd> and typing the name of the relevant command. This is the recommended way to use the
```
Comment (mlutze) [inline src/content/BuildAndPackages.js]: Is this key combination the same on all OSes?

Comment (mlutze) [reply src/content/BuildAndPackages.js]: For me (Ubuntu) it is `CTRL-SHIFT-P`

File: src/content/BuildAndPackages.js [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+                        build system.
+                    </p>
+
+                    <p>
+                        <b>Tip:</b> To create a new project in Visual Studio Code:
+                        <ol>
+                            <li> create a new empty folder,</li>
+                            <li>open the folder in Visual Studio Code (File -> Open Folder...), and</li>
+                            <li>press <kbd>CTRL + P</kbd> and type <code>Flix init</code>.
```
Comment (mlutze) [inline src/content/BuildAndPackages.js]: see above

File: src/content/BuildAndPackages.js [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+                    <p>
+                        The JAR-file contains all class files from the <Code>build/</Code> directory.
+                    </p>
+
+                    <Warning>
+                        The project must have been built before with the <Code>build</Code> command.
+                    </Warning>
+
+                    <DesignNote>
+                        At the time of writing, the built JAR-file still depends on the <Code>flix.jar</Code> file. Thus
```
Comment (mlutze) [inline src/content/BuildAndPackages.js]: Would it be useful to put a real (approximate) date here, so people know how up-to-date this documentation is?

File: src/content/BuildAndPackages.js [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+                        which will produce a <Code>myproject.fpkg</Code> package.
+                    </p>
+
+                    <p>
+                        A Flix package file is essentially a zip-file of the project source code. A Flix package file
+                        can be reused in another project by placing it into the <Code>lib/</Code> directory.
+                    </p>
+
+                    <p>
+                        Its recommended to include the semantic version in the filename of the package,
```
Comment (mlutze) [inline src/content/BuildAndPackages.js]: It's* :)

File: src/content/BuildAndPackages.js [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+                    <CodeBlock>
+                        {`$ java -jar myproject.jar`}
+                    </CodeBlock>
+
+                    <p>
+                        The JAR-file contains all class files from the <Code>build/</Code> directory.
+                    </p>
+
+                    <Warning>
+                        The project must have been built before with the <Code>build</Code> command.
```
Comment (mlutze) [inline src/content/BuildAndPackages.js]: "beforehand" would be a little more clear

**Pull request merged**